cmd/antrea-agent/options.go

```diff
@@ -397,10 +396,18 @@ func (o *Options) setK8sNodeDefaultOptions() {
 	if o.config.HostProcPathPrefix == "" {
 		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
 	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		o.config.AntreaProxy.Enable = pointer.Bool(false)
+		klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
+	} else if o.config.AntreaProxy.Enable == nil {
+		o.config.AntreaProxy.Enable = pointer.Bool(true)
+	}
```

Comment:
I feel it may be more clear to seperate defaulting and validating.
When defaulting, we just ensure `o.config.AntreaProxy.Enable` is set to `true` if it's not already set.
When validating, we alert if `features.DefaultFeatureGate.Enabled(features.AntreaProxy)` is false, and only enable antreaProxy when both of them are true.

Reply:
Make sense.